diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SingleLineStatementsRule.kt

```diff
@@ -0,0 +1,98 @@
+        const val minListTextSize = 1
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        checkSemicolon(node)
+    }
+
+    private fun checkSemicolon(node: ASTNode) {
+        node.getChildren(semicolonToken).forEach {
+            if (!isSemicolonAtEndOfLine(it)) {
+                if (isError(it)) {
+                    MORE_THAN_ONE_STATEMENT_PER_LINE.warnAndFix(configRules, emitWarn, isFixMode, findWrongText(it),
+                            it.startOffset) {
+                        if (it.treeParent.elementType == ENUM_ENTRY){
+                            node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeNext)
+                        } else {
+                            if (!it.isBeginByNewline()) {
+                                val nextNode = it.treeNext ?: it.parent({ parent -> parent.treeNext != null})
+                                if ( nextNode != null && nextNode.elementType == WHITE_SPACE)
+                                    (nextNode as LeafPsiElement).replaceWithText("\n")
+                                else
+                                    node.addChild(PsiWhiteSpaceImpl("\n"), it)
+                            }
+                            node.removeChild(it)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Sometimes the semicolon is the last leaf of the tree at a given level, so you need to go up a few levels to check if there is something behind the semicolon
+     */
+    private fun isError (node: ASTNode) = node.parent({ it.treeNext != null}, strict=false)?.let { !it.isFollowedByNewline() } ?: false
+
+
+    private fun isSemicolonAtEndOfLine(node: ASTNode) = (node.treeNext != null && node.isFollowedByNewline())
+
+    private fun findWrongText(node: ASTNode): String {
+        var text: MutableList<String> = mutableListOf()
+        var prevNode: ASTNode? = node.treePrev
+        var nextNode: ASTNode? = node
+        while (prevNode != null) {
+            val listText = prevNode.text.split("\n")
+            text.add(listText.last())
+            if (listText.size > minListTextSize)
+                break
+            prevNode = prevNode.treePrev
+        }
+        text = text.asReversed()
+        text.add(node.text)
+        nextNode = nextNode?.parent({ it.treeNext != null }, false) ?: nextNode
```

Comment:
```suggestion
        var nextNode = node?.parent({ it.treeNext != null }, false) ?: node
```
I suggest to declare `nextNode` here, because you don't use it before.